File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer,Integer> numsMap = new HashMap<>();
        Integer index;
        for( int i = 0 ; i < nums.length ; i++ ){
            index = numsMap.get(target-nums[i]);
            if( index != null ){
                result[0] = Integer.min(i,index);
                result[1] = Integer.max(i,index);
                break;
            }
            numsMap.put(nums[i],i);
        }
        return result;  
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow = 0
        fast = 0
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                break
        i = 0
        while i != slow :
            i = nums[i]
            slow = nums[slow]                                
        return i
         